### ISSUE TITLE
URL query param results in parser error in Firefox

### ISSUE THREAD

**Issue by iamtheddrman** [swagger-api/swagger-ui#5055]
### Q&A
 - OS: Windows 10 and Ubuntu
 - Browser: Firefox
 - Version: 63
 - Method of installation: dist assets
 - Swagger-UI version: 3.20.1 (exists in older versions too)
 - Swagger/OpenAPI version: Swagger 2.0

### Content & configuration
Example Swagger/OpenAPI definition:
```yaml
swagger: '2.0'
info:
  description: This is a description
  title: Test For Swagger Bug
  contact:
    name: No One
    url: 'http://www.notarealsite.com'
    email: Help@notarealsite.com
  license:
    name: Copyright No One 2018.
  version: '1.0'
paths:
  /test:
    get:
      schemes:
        - ws
      tags:
        - Info
      summary: Creates a websocket
      operationId: testSocket
      responses:
        '200':
          description: Socket established
        '400':
          description: Bad request
          schema:
            type: string
        '500':
          description: Socket error
          schema:
            type: string
definitions:
  Test:
    type: object
    properties:
      prop:
        type: array
        items:
          type: integer
          format: uint8
    x-go-package: test/mypkg
```

Swagger-UI configuration options:
```
?url=/path/to/above/file.yaml
```

### Describe the bug you're encountering
When using the ?url query parameter to populate the contents of the Swagger UI, the above YAML file results in a parser error "The stream contains non-printable characters. However, copy/pasting the exact same file into Swagger Editor renders the file correctly.

### To reproduce...
1. Save the above YAML to a file (I used Unix line endings)
2. Serve up the file with something like Python's SimpleHTTPServer
3. Use Firefox to open the online Swagger UI (http://petstore.swagger.io/) or a local copy of it, adding the URL query param

### Expected behavior
The file works correctly in Chrome and Edge, but fails in Firefox. I would expect it to behave the same across all major browsers.

### Additional context or thoughts
As mentioned above, this has been an issue for some time. I have a much older version of Swagger UI copied down locally (not sure of the exact version, but most likely early 2017). If someone can direct me on how to determine the exact version of Swagger UI from the dist folder contents, I'll be glad to do so.

Also, if there's a workaround to get this working in the meantime, that would be much appreciated. I have a custom version of the Swagger UI with buttons that load our various swagger specs via the ?url query param when clicked. It works for Chrome, but the fact that it doesn't for Firefox could be an issue for our end users.

**Comment by shockey** [comment]
Hi @iamtheddrman, I'm not able to reproduce this in Firefox 63 on my Mac:

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/680248/49380179-48a3d000-f711-11e8-9a3b-6b998e3c0281.png">
 
Can you upload the file directly from your filesystem somewhere (without copy/pasting)? Pasting it into the GitHub issue may have stripped the offending content.

**Comment by iamtheddrman** [comment]
@shockey Thanks for the response. I have not tested on a Mac, but can reproduce the problem with both a Windows 10 machine and Ubuntu machine even copy/pasting the contents of this bug report into a new file.

My original file is here: https://ufile.io/le7t4
And the new file copy/pasted from the contents of the bug is here: https://ufile.io/r4xf6

Both links will expire in 30 days, but hopefully this will be resolved before then. I can re-upload if needed.

I've also tried changing line endings (Windows and Unix) as well as the indention (2 or 4 spaces). I get the same result in every case (works in Chrome/Edge, fails in Firefox).

I would be very happy if there was something I've simply done wrong here, but since it works in Chrome and Edge, I'm concerned that it may not be the case. If so, I will gladly update my files to ensure I don't hit this issue again.

**Comment by shockey** [comment]
Hi @iamtheddrman, apologies for the delay here - we've been a bit swamped with the issue tracker over this holiday season. Can you upload those assets again? I'll pull them down as soon as you do, and take a look.

**Comment by iamtheddrman** [comment]
Sure. The copy/pasted file is here: https://ufile.io/4jruh
And the original is here: https://ufile.io/bfj1d

Hopefully you can find something that I've been unable to! I appreciate you looking into the issue.

**Comment by mikedilger** [comment]
Not to hijack, but I'm experiencing a very similar issue with query parameters under Firefox (67.0b14) which render as "😱 Could not render this component, see the console."

The console says
````
Warning: escaping deep link whitespace with `_` will be unsupported in v4.0, use `%20` instead. layout.js:99:14
TypeError: "O is undefined"
    value parameter-row.jsx:149
    render root-injects.jsx:93
    React 9
root-injects.jsx:95:14
````

If this is a different issue, let me know and I'll open it separately.

I have a number of security/privacy tweaks to my firefox, and one of them could well be causing this. I haven't tried with a clean out-of-the-box profile.

The parameters were simply:
````        "parameters": [
          {
            "name": "username",
            "in": "query",
            "required": true,
            "type": "string",
            "description": "Your username"
          }````

**Comment by mikedilger** [comment]
After disabling all of my plugins the problem went away. After re-enabling the plugins, the problem didn't come back. Weird.

**Comment by iamtheddrman** [comment]
Has anyone looked further into this issue? I know the links are bad again, but I don't think the files themselves are really the issue. We just had another customer using Firefox call our support line over this issue. None of our customers use Mac OS X, but many use Windows and Linux (as noted in the original bug description) and we're consistently seeing this problem on both of those OSes.

**Comment by iamtheddrman** [comment]
We were finally able to track down the issue and it is indeed unrelated to the Swagger UI. Our web server was modified to start serving files compressed with gzip at a customer's request, but we did not verify that the way they were being served was compatible with Swagger UI. I have added an exception to the compression for Swagger-related files and everything is working appropriately with all browsers. Thanks for the time, and my apologies for wasting it.
